pkg/reconciler/taskrun/resources/taskref_test.go

```diff
@@ -237,10 +264,19 @@ func TestLocalTaskRef(t *testing.T) {
 				Tektonclient: tektonclient,
 			}
 
+			if tc.localTaskRefResolverWithoutNamespace {
+				lc.Namespace = ""
+			}
+
 			task, refSource, _, err := lc.GetTask(ctx, tc.ref.Name)
-			if tc.wantErr && err == nil {
-				t.Fatal("Expected error but found nil instead")
-			} else if !tc.wantErr && err != nil {
+			if tc.wantErr != nil {
+				if err == nil {
+					t.Fatal("Expected error but found nil instead")
+				}
+				if tc.wantErr.Error() != err.Error() {
+					t.Fatalf("Received different error ( %#v )", err)
```

Comment:
is it a typo (%#v)? We can also print the expected error to make it easier to debug.

Reply:
I think %#v is a[ Go-syntax representation of the value ](https://pkg.go.dev/fmt) and this was previously used at https://github.com/tektoncd/pipeline/blob/5e93dd313fa7aa82029a5f33fc20b4b4de9df9ba/pkg/reconciler/taskrun/resources/taskref_test.go#L244.

Reply:
I personally lean towards that we should expect to change the error messages in case where we have changed the logics of some errors and there could be changes in sequence of errors being emitted.

Reply:
In that case, can we use cmp.Diff to compare the expected error to the received error?